pyfr/plugins/ascent.py

```diff
@@ -0,0 +1,441 @@
+# -*- coding: utf-8 -*-
+
+from collections import defaultdict
+from ctypes import (c_char_p, c_double, c_float, c_long, c_longlong, c_void_p,
+                    RTLD_GLOBAL)
+import re
+
+import numpy as np
+
+from pyfr.ctypesutil import LibWrapper
+from pyfr.mpiutil import get_comm_rank_root
+from pyfr.nputil import npeval
+from pyfr.plugins.base import BasePlugin, RegionMixin
+from pyfr.shapes import BaseShape
+from pyfr.util import file_path_gen, file_path, subclass_where
+from pyfr.writers.vtk import BaseShapeSubDiv
+
+
+class AscentError(Exception): pass
+class ConduitError(Exception): pass
+
+
+class ConduitWrappers(LibWrapper):
+    _libname = 'conduit'
+    _mode = RTLD_GLOBAL
+
+    # Functions
+    _functions = [
+        (c_void_p, 'conduit_node_append', c_void_p),
+        (c_void_p, 'conduit_node_create', c_void_p),
+        (None, 'conduit_node_destroy', c_void_p),
+        (c_void_p, 'conduit_node_fetch', c_void_p, c_char_p),
+        (c_void_p, 'conduit_node_set_path_char8_str', c_void_p, c_char_p,
+         c_char_p),
+        (c_void_p, 'conduit_node_set_path_float32', c_void_p, c_char_p,
+         c_float),
+        (c_void_p, 'conduit_node_set_path_float32_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_float64', c_void_p, c_char_p, 
+         c_double),
+        (c_void_p, 'conduit_node_set_path_float64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_int32', c_void_p, c_char_p, c_long),
+        (c_void_p, 'conduit_node_set_path_int64_ptr', c_void_p, c_char_p,
+         c_void_p, c_longlong),
+        (c_void_p, 'conduit_node_set_path_node', c_void_p, c_char_p, c_void_p),
+        (c_void_p, 'conduit_node_print', c_void_p),
+    ]
+
+    # Errors
+    _statuses = {'*': ConduitError}
```

Comment:
No need here as we have our own custom errcheck we can just raise ConduitError directly.